core/trino-spi/src/main/java/io/trino/spi/ptf/TableFunctionAnalysis.java

```diff
@@ -50,6 +62,11 @@ public Optional<Descriptor> getReturnedType()
         return returnedType;
     }
 
+    public Map<String, Set<String>> getRequiredColumns()
```

Comment:
What are the semantics when:
* A table argument is missing from the map
* The set for a given table argument is empty

Also, is the list or required columns the minimum required set of columns or the exact set of required columns (i.e., are any columns not mentioned optionally used by the function or not used at all)?

Reply:
> What are the semantics when:
A table argument is missing from the map
The set for a given table argument is empty

According to the spec, the `describe` procedure _must_ request a non-empty set of columns from every input table. If there's no `describe` procedure, the requested columns set defaults to the full relation type for each input table.

In our implementation, we can handle it more granularly, that is, per input table. E.g. if there's no entry in the map for a table argument, we could assume that all columns are required. That brings the question of accessing the columns by name (mapping names to channels) at compilation time: should we only allow to access names that were explicitly declared during the analysis, or all column names?

However, to follow the spec more closely, we should require that there is an entry in the map for each table argument, and that it is not empty. But then we lose the possibility ro require _all_ columns in the case when some columns are anonymous.

Reply:
>Also, is the list or required columns the minimum required set of columns or the exact set of required columns (i.e., are any columns not mentioned optionally used by the function or not used at all)?

This is the exact set of required columns. There will be column pruning based on these declarations.